test/test_rw_fortran.F90

```diff
@@ -375,27 +376,38 @@ END PROGRAM main
 
 ! Generate a simple, 1D sinusioidal data array with some noise
 SUBROUTINE gen_data(npoints, noise, amp, buf)
-  
   USE ISO_C_BINDING
+  USE ISO_FORTRAN_ENV, ONLY: REAL128
   IMPLICIT NONE
   INTEGER(C_SIZE_T) :: npoints
   REAL(C_DOUBLE) :: noise
   REAL(C_DOUBLE) :: amp
   REAL(C_DOUBLE), DIMENSION(1:npoints) :: buf
 
+  REAL(REAL128), PARAMETER :: PI = 3.1415926535897932384626433832795028841971_REAL128
```

Comment:
What if the platform does not support 128? Does not this cause an error? What benefit is this over just making it a double and over-prescribe the precision?

Reply:
That is a valid point. According to the 2008 Fortran Standard the REAL128 will have a negative value if the target platform does not support it. Let me change it to REAL64, assuming that this is defined for all platforms. This also means that we can only specify PI up to 12 digits after the period.

Reply:
I would just make it C_DOUBLE. Fortran does not complain about extra digits, it will just truncate the number.

Reply:
That is a good idea.